OSDOCS-2455: fsGroup and pod timeouts (#41415)
openshift/openshift-docs: modules/storage-persistent-storage-fsGroup.adoc

// Module included in the following assemblies:
//
// * storage/understanding-persistent-storage.adoc

[id="using_fsGroup_{context}"]
= Using fsGroup to reduce pod timeouts

> If a storage volume contains many files (~1,000,000 or greater), you may experience pod timeouts.

Reviewer: Would be possible to eliminate the parenthetical if this were a specific technical limit, e.g. "If a storage volume contains more than X files..."

> This can occur because, by default, {product-title} recursively changes ownership and permissions for the contents of each volume to match the `fsGroup` specified in a pod's `securityContext` when that volume is mounted. For large volumes, checking and changing ownership and permissions can be time consuming, slowing pod startup. You can use the `fsGroupChangePolicy` field inside a `securityContext` to control the way that {product-title} checks and manages ownership and permissions for a volume.

Reviewer: Is "This" == "Timeouts?"

Might want to rephrase to aid translation.

> This can occur because, by default, {product-title} recursively changes ownership and permissions for the contents of each volume to match the `fsGroup` specified in a pod's `securityContext` when that volume is mounted. For large volumes, checking and changing ownership and permissions can be time consuming, slowing pod startup. You can use the `fsGroupChangePolicy` field inside a `securityContext` to control the way that {product-title} checks and manages ownership and permissions for a volume.

Reviewer: ```suggestion
This can occur because, by default, {product-title} recursively changes ownership and permissions for the contents of each volume to match the `fsGroup` that is specified in a pod's `securityContext` [$NOUN] when that volume is mounted. For large volumes, checking and changing ownership and permissions can be time consuming, slowing pod startup. You can use the `fsGroupChangePolicy` field inside a `securityContext` to control the way that {product-title} checks and manages ownership and permissions for a volume.
```

> `fsGroupChangePolicy` defines behavior for changing ownership and permission of the volume before being exposed inside a pod. This field only applies to volume types that support `fsGroup`-controlled ownership and permissions. This field has two possible values:

Reviewer: ```suggestion
The `fsGroupChangePolicy` field defines behavior for changing ownership and permission of the volume before being exposed inside a pod. This field only applies to volume types that support `fsGroup`-controlled ownership and permissions. This field has two possible values:
```

Again, thinking of translation here. 

"being" confuses me a bit. Before the volume is exposed?

> * `OnRootMismatch`: Only change permissions and ownership if permission and ownership of root directory does not match with expected permissions of the volume. This can help shorten the time it takes to change ownership and permission of a volume to reduce pod timeouts.

Reviewer: ```suggestion
* `OnRootMismatch`: Only change permissions and ownership if permission and ownership of the root directory does not match with the expected permissions of the volume. This behavior [?] can help shorten the time it takes to change ownership and permission of a volume to reduce pod timeouts.
```

* `Always`: Always change permission and ownership of the volume when a volume is mounted.

.`fsGroupChangePolicy` example
[source,yaml]
----
securityContext:
  runAsUser: 1000
  runAsGroup: 3000
  fsGroup: 2000
  fsGroupChangePolicy: "OnRootMismatch" <1>
  ...
----

> <1> `OnRootMismatch` specifies skipping recursive permission change, thus helping to avoid pod timeout problems.

Reviewer: ```suggestion
<1> `The OnRootMismatch` value causes your cluster to skip recursive permission changes, thus reducing pod timeout problems.
```

Just a thought to simplify the grammar.

[NOTE]
====

> The fsGroupChangePolicyfield has no effect on ephemeral volume types, such as secret, configMap, and emptydir.

Reviewer: Do these need to be in backticks?
